src/component/TodoList.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import binImage from '../assets/icons/bin.png';
+
+function TodoList() {
+  const [inpuValue, setInputValue] = useState('');
+  const [array, setArray] = useState([]);
+
+  function addTodo() {
+    if (inpuValue) {
+      const object = {
+        value: inpuValue,
+        completed: false,
+      };
+      setArray([...array, object]);
+      localStorage.setItem('todo-list', JSON.stringify([...array, object]));
+    }
+  }
+
+  useEffect(() => {
+    const getItem = JSON.parse(localStorage.getItem('todo-list'));
+    if (getItem !== null) {
+      setArray(getItem);
+    }
+  }, []);
+
+  function ischecked(e) {
+    const newArray = [];
+    for (let i = 0; i < array.length; i += 1) {
+      if (i === parseInt(e.target.parentElement.classList[0], 10)) {
+        newArray.push(array[i]);
+        if (!newArray[i].completed) {
+          newArray[i].completed = true;
+        } else {
+          newArray[i].completed = false;
+        }
+      } else {
+        newArray.push(array[i]);
+      }
+    }
+    setArray(newArray);
+    localStorage.setItem('todo-list', JSON.stringify(newArray));
+  }
+
+  function delet(e) {
```

Comment:
- [Optional] Consider adding more descriptive names to your functions like "deleteTask" as it can make your code more readable and easier to understand